assets/training/aoai/proxy_components/src/finetune_submit.py

```diff
@@ -40,27 +46,72 @@ def submit_finetune_job(self, training_file_id, validation_file_id, model, regis
             validation_file=validation_file_id,
             hyperparameters=hyperparameters,
             suffix=suffix)
-        job_id = finetune_job.id
+        self.job_id = finetune_job.id
         status = finetune_job.status
 
-        # If the job isn't yet done, poll it every 30 seconds.
-        if status not in ["succeeded", "failed"]:
+        terminal_statuses = ["succeeded", "failed", "cancelled"]
+        # If the job isn't yet done, poll it every 60 seconds.
+        if status not in terminal_statuses:
             logger.info(f'Job not in terminal status: {status}. Waiting.')
-            while status not in ["succeeded", "failed"]:
-                time.sleep(30)
-                finetune_job = self.aoai_client.fine_tuning.jobs.retrieve(job_id)
-                status = finetune_job.status
-                logger.info(f"current status of finetune job : {status}, polling after 30 sec")
-        else:
-            logger.debug(f'Fine-tune job {job_id} finished with status: {status}')
+            last_event = None
+            while status not in terminal_statuses:
+                time.sleep(10)
 
-        if status != "succeeded":
-            raise Exception("Component failed")
-        else:
-            finetuned_model_id = finetune_job.fine_tuned_model
-            logger.info(f'Fine-tune job {job_id} finished with \
-                        status: {status}. model id: {finetuned_model_id}')
-            return finetuned_model_id
+                finetune_job = self.aoai_client.fine_tuning.jobs.retrieve(self.job_id)
+                status = finetune_job.status
+                last_event = self._log_events(last_event)
+
+        if status == "failed":
+            error = finetune_job.error
+            logger.error(f"Fine tuning job: {self.job_id} failed with error: {error}")
+            raise Exception(f"Fine tuning job: {self.job_id} failed with error: {error}")
+
+        elif status == "cancelled":
+            logger.info(f"finetune job: {self.job_id} got cancelled")
+            return None
+
+        """metrics can be retrived only for successful finetune job"""
+        finetuned_model_id = finetune_job.fine_tuned_model
+        logger.info(f'Fine-tune job: {self.job_id} finished successfully. model id: {finetuned_model_id}')
+        logger.info("fetching training metrics from Azure OpenAI")
+        self._log_metrics(finetune_job)
+        return finetuned_model_id
+
+    def cancel_job(self):
+        """Cancel finetuning job in Azure OpenAI resource."""
+        logger.debug("job cancellation has been triggered, cancelling finetuning job")
+        return self.aoai_client.fine_tuning.jobs.cancel(self.job_id)
```

Comment:
check whether self.job_id is not None?

Reply:
self.job_id can be none in case finetune job has not been submitted yet. In that case exit() is called and it is handled gracefully in the main function